examples/torch_ddp_benchmark.py

```diff
@@ -0,0 +1,305 @@
+# Adapted from https://github.com/pytorch/pytorch/tree/main/benchmarks/distributed/ddp
```

Comment:
thanks for adding this! The changes here are useful since I couldn't get original example to work out of the box. Can we add quick comments on why the changes here are necessary?

Reply:
ok i thought i might be the only one that couldn't get the example immediately to work. it's missing `--local-rank` and the `--distributed-backend` doesn't actually do anything in the script. it's nice moving to torchrun. resolving.